python/pyarrow/tests/test_compute.py

```diff
@@ -860,6 +860,16 @@ def test_fill_null():
     expected = pa.array([None, None, None, None])
     assert result.equals(expected)
 
+    arr = pa.array(['a', 'bb', None])
+    result = arr.fill_null('ccc')
+    expected = pa.array(['a', 'bb', 'ccc'])
+    assert result.equals(expected)
+
+    arr = pa.array([b'a', b'bb', None], type=pa.large_binary())
+    result = arr.fill_null('ccc')
```

Comment:
Hmm, we can pass a unicode string here? Interesting.

Reply:
When writing it, I was doing it as a check to the system, expecting it to fail, and forget to change this. However, thinking about it, I'm not sure how it should be checked since we have both LargeStringScalar and StringScalar.

Reply:
Yes, it's probably ok.